### PR TITLE
[onkyo] Add channels for media info

### PR DESCRIPTION
**Pull request by marcelrv**
Add channels for audio & video in & out information similar as in the onkyo app

Signed-off-by: Marcel Verpaalen <marcel@verpaalen.com>

### CODE REVIEW

**Comment by lolodomo** [comment]
Please update README.

**Comment by lolodomo** [comment]
Please also run the maven command to update your default i18n properties file

File: bundles/org.openhab.binding.onkyo/src/main/java/org/openhab/binding/onkyo/internal/OnkyoParserHelper.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+ * @author Marcel Verpaalen - Initial contribution
+ */
+@NonNullByDefault
+public final class OnkyoParserHelper {
+
+    public static State infoBuilder(String data, int from, int to) {
+        StringBuilder builder = new StringBuilder();
+        int end = to;
+        String[] element = data.split(",");
+        if (element.length < from) {
```
Comment (lolodomo) [inline bundles/org.openhab.binding.onkyo/src/main/java/org/openhab/binding/onkyo/internal/OnkyoParserHelper.java:31]: Please check but I believe it should be `<=`

File: bundles/org.openhab.binding.onkyo/src/main/java/org/openhab/binding/onkyo/internal/OnkyoParserHelper.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+            return StringType.EMPTY;
+        }
+        if (element.length < end) {
+            end = element.length;
+        }
```
Comment (lolodomo) [inline bundles/org.openhab.binding.onkyo/src/main/java/org/openhab/binding/onkyo/internal/OnkyoParserHelper.java:36]: Do you want to include or exclude "end" index ?

File: bundles/org.openhab.binding.onkyo/src/main/java/org/openhab/binding/onkyo/internal/OnkyoParserHelper.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        boolean firstElement = true;
+        for (int i = from; i < end; i++) {
+            if (!element[i].isEmpty() && !firstElement) {
+                builder.append(", ");
+            }
+            builder.append(element[i]);
+            firstElement = false;
+        }
```
Comment (lolodomo) [inline bundles/org.openhab.binding.onkyo/src/main/java/org/openhab/binding/onkyo/internal/OnkyoParserHelper.java:44]: You can do this with less lines of code.
Here is an example I used in one of my recent PR (after I got the same comment :) ). You can adapt it to your need.
```
        Object @Nullable [] params = msgParams;
        if (params != null && params.length > 0) {
            result += Stream.of(params).map(param -> String.format("\"%s\"", param == null ? "" : param.toString()))
                    .collect(Collectors.joining(", ", " [ ", " ]"));
        }
```

File: bundles/org.openhab.binding.onkyo/src/main/resources/OH-INF/thing/channels.xml [hunk 4]
```diff
@@ -188,5 +188,29 @@
...
+		<item-type>String</item-type>
+		<label>Audio Output Info</label>
+		<description>Audio Output Stream Information</description>
+		<state readOnly="true"></state>
+	</channel-type>
+	<channel-type id="videoIn" advanced="true">
+		<item-type>String</item-type>
+		<label>Video Input Info</label>
+		<description>Video Input Stream Information</description>
+		<state readOnly="true" pattern="%s"></state>
```
Comment (lolodomo) [inline bundles/org.openhab.binding.onkyo/src/main/resources/OH-INF/thing/channels.xml:207]: Is there a reason we specify a pattern for some of the new channels but not all ?

**Review by lolodomo — CHANGES_REQUESTED** [review]

File: bundles/org.openhab.binding.onkyo/src/main/resources/OH-INF/thing/channels.xml (hunk 4 above)
Comment (marcelrv) [reply bundles/org.openhab.binding.onkyo/src/main/resources/OH-INF/thing/channels.xml:207]: None at all...  Did not notice it... 
I wrote the code 5years ago... and needed to download the  onkyo app just for this info and remembered I did write this way back. time to dust if off

File: bundles/org.openhab.binding.onkyo/src/main/java/org/openhab/binding/onkyo/internal/OnkyoParserHelper.java (hunk 3 above)
Comment (marcelrv) [reply bundles/org.openhab.binding.onkyo/src/main/java/org/openhab/binding/onkyo/internal/OnkyoParserHelper.java]: yes, sure can. 
I'm still not too familiar with labda/streams so it is not my typical first way to go... though it is indeed quite compact.

**Review by lolodomo — APPROVED** [review]
LGTM, thanks

**Pull request merged**